Libraries/Opc.Ua.Server/Diagnostics/DiagnosticsNodeManager.cs

```diff
@@ -1401,13 +1486,57 @@ private bool UpdateServerDiagnosticsSummary()
                         UpdateSubscriptionDiagnostics(m_subscriptions[ii], subscriptionArray, ii);
                     }
 
+                    // filter out the members which corespond to users that are not allowed to see their contents
+                    SubscriptionDiagnosticsDataType[] selectedToPass = new SubscriptionDiagnosticsDataType[m_sessions.Count];
```

Comment:
this loop is duplicated three times in this elseif block. private helper?

Reply:
moved all the filtering in the above loops